internal/pkg/pkg.go

```diff
@@ -436,6 +466,117 @@ func Subpackages(rootPath string, matcher SubpackageMatcher, recursive bool) ([]
 	return paths, nil
 }
 
+// SubpackagesWithFs returns a slice of paths to any subpackages of the provided path.
+// The matcher parameter decides if all types of subpackages should be considered,
+// and the recursive parameter determines if only direct subpackages are
```

Comment:
Reading this I wonder about the distinction between direct subpackages vs. some other kind of subpackages; both seem 'recursive' in nature; we may want to change the name of the recursive flag to clearly capture its purpose.

IMO recursive == descend into subpackages (of any kind)

Reply:
Matcher is to select Remote vs Local subpackages. recursive flag is related to depth. Updated the description about Matchers.